src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -37,14 +38,19 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                var requestTelemetry = new RequestTelemetry
+                var requestTelemetry = new RequestTelemetry()
                 {
                     Id = httpContext.TraceIdentifier
                 };
-
                 requestTelemetry.Start(/*timestamp*/);
-
                 httpContext.Features.Set(requestTelemetry);
+
+                this.client.Initialize(requestTelemetry);
```

Comment:
Could you put the call to ```Intialize``` before the call to ```Start```? This is because of https://github.com/Microsoft/ApplicationInsights-dotnet/pull/431 (waiting to be merged). ```Start``` now sends ETW events and the payload includes Operation ID and Parent Operation ID which may be set in initializers.